src/components/markdown_editor/markdown_editor.tsx

```diff
@@ -72,6 +72,16 @@ type CommonMarkdownEditorProps = Omit<
     /** ID of an element describing the text editor, useful for associating error messages */
     'aria-describedby'?: string;
 
+    /**
+     * Optional class applied to the textarea element
+     */
+    textareaClassName?: string;
+
+    /**
+     * Optional class applied to the preview element
+     */
+    previewClassName?: string;
```

Comment:
I would just extend these to be general props based on the element they get applied to. So that instead of restricting to only allowing custom classes, they could pass ID's, data-test-subj, etc. 
```suggestion
    textareaProps?: TextareaHTMLAttributes<HTMLTextAreaElement> & CommonProps;

    /**
     * Optional props applied to the preview element
     */
    previewProps?: <HTMLAttributes<HTMLDivElement> & CommonProps;
```

Reply:
That's a good idea. Thanks.